apps/examples/testcase/le_tc/kernel/tc_smart_sfs.c

```diff
@@ -0,0 +1,895 @@
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_5byte
+ * @brief                :reading first few bytes of the input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_5byte(void)
+{
+	char *input = "abcdefgh12345678abcd";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 1, 5);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input, 5);
+	TC_ASSERT_EQ("tc_readtest_5byte", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_middlebyte
+ * @brief                :reading a single byte from the middle of input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_middlebyte(void)
+{
+	char *input = "abcdefgh12345678abcdefgh";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 11, 1);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input + 10, 1);
+	TC_ASSERT_EQ("tc_readtest_middlebyte", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_middlebyte10
+ * @brief                :reading a few bytes from the middle of input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_middlebyte10(void)
+{
+	char *input = "abcdefgh12345678abcdefgh12345678";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 20, 10);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input + 19, 10);
+	TC_ASSERT_EQ("tc_readtest_middlebyte10", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+void smart_sfs_main_util(void)
+{
+	tc_fwrite_data2();
+	tc_fwrite_data8();
+	tc_fwrite_data16();
+	tc_fwrite_data24();
+	tc_fwrite_data121();
+	tc_fwrite_data255();
+	tc_fprintf_data2();
+	tc_fprintf_data8();
+	tc_fprintf_data16();
+	tc_fprintf_data24();
+	tc_fprintf_data121();
+	tc_fprintf_data255();
+	tc_fputs_data2();
+	tc_fputs_data8();
+	tc_fputs_data16();
+	tc_fputs_data24();
+	tc_fputs_data121();
+	tc_fputs_data255();
+	tc_fgets_data2();
+	tc_fgets_data8();
+	tc_fgets_data16();
+	tc_fgets_data24();
+	tc_fgets_data121();
+	tc_fgets_data255();
+	tc_fgets_fprintf_data2();
+	tc_fgets_fprintf_data8();
+	tc_fgets_fprintf_data16();
+	tc_fgets_fprintf_data24();
+	tc_fgets_fprintf_data121();
+	tc_fgets_fprintf_data255();
+	tc_readtest_1byte();
+	tc_readtest_5byte();
+	tc_readtest_middlebyte();
+	tc_readtest_middlebyte10();
+}
+
+int smart_sfs_main(void)
+{
+	pthread_t p;
+	pthread_attr_t attr;
+
+	pthread_setname_np(p, SFS_thread_name);
+	pthread_attr_init(&attr);
+	pthread_attr_setstacksize(&attr, SFS_TASK_STACKSIZE);
+	pthread_create(&p, &attr, (pthread_startroutine_t)smart_sfs_main_util, NULL);
```

Comment:
You must use pthread_join or pthread_detach after creating pthread. It causes memeory leak.